### DISCUSSION TITLE
Scan only files in a pull request (Azure DevOps)

### DISCUSSION THREAD

**Discussion by lfaltran** [JetBrains/Qodana#168] [Q&A]
Hello everyone.

I'm working with Qodana on Azure DevOps pipeline, and looking at configs for other CI platforms I've noticed that GitHub Actions has a "pr-mode" param (https://github.com/JetBrains/qodana-action#configuration)

Is there any option on Azure DevOps or an alternative to speed up the code analysis? (I'm working already with caches)

**Comment by brichbash** [comment]
Hello,

We don't have one yet. However, I've created a feature request [on our tracker](https://youtrack.jetbrains.com/issue/QD-5793/Add-pr-mode-to-Qodana-Azure-extension). If you have a spare minute, please vote for it to get notified when there are any updates.

**Comment by lfaltran** [comment]
Hey @brichbash thank you for your support ;)

**Comment by martythewizard** [comment]
Greetings I am wondering where this discussion is currently at. Is it still being looked as a possible implementation?

**Reply by shaunhavelaarx** [reply]
I don't understand why this isn't getting attention, it's the main reason why companies want static code analysis. It's the only feature that's preventing us from going forward with qodana

**Comment by brichbash** [comment]
Hello,

Although there is still no built-in feature for analyzing PRs in the Azure extension like in our GitHub Action, if you want to analyze only changes in PRs on Azure, you can use the recently introduced `--diff-start <commit-hash>` flag. [Here](https://www.jetbrains.com/help/qodana/analyze-pr.html#cgrluc_34) is an example for GitLab with the usage. You can pass a variable to the flag that contains the commit hash for the HEAD of the target branch of the pull request
